File: JavaScript/searching/linear_search.js
const linearSearch = (arr, targetVal) => {
  // loops through the array
  for (let x in arr) {
    // checks for a match
    if (arr[x] === targetVal) {
      return arr.indexOf(arr[x]);
    }
  }
  // if no match is found
  return -1;
};

const arr = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
const targetVal = 7;
const result = linearSearch(arr, targetVal);
if (result != -1) {
  console.log("Value", targetVal, "found at index", result);
} else {
  console.log("Value", targetVal, "not found");
}

File: searching algorithms/linear search/linear_search.py
def linearSearch(arr, targetVal):
    for i in range(len(arr)):
        if arr[i] == targetVal:
            return i
    return -1


arr = [3, 7, 2, 9, 5]
targetVal = 9

result = linearSearch(arr, targetVal)

if result != -1:
    print("Value", targetVal, "found at index", result)
else:
    print("Value", targetVal, "not found")
